features/command.feature

```diff
@@ -1579,3 +1579,39 @@ Feature: WP-CLI Commands
       """
       core custom-subcommand
       """
+
+  Scenario: Custom core subcommand should be able to run on before_wp_load
+    Given a WP installation
+    And a wp-cli.yml file:
+      """
+      require:
+        - core-safe-update.php
+      """
+    And a core-safe-update.php file:
+      """
+      <?php
+      /**
+       * Safely update WP core.
+       *
+       * ## OPTIONS
+       *
+       * [--version=<version>]
+       * : Specify a custom WordPress version.
+       *
+       * @when before_wp_load
+       */
+      function wpx_core_safe_update() {
+        WP_CLI::success( 'Core safely updated.' );
+      }
+      WP_CLI::add_command( 'core safe-update', 'wpx_core_safe_update' );
+      // Da heck?
+      WP_CLI::add_hook( 'find_command_to_run_pre', function() {
```

Comment:
@schlessera Any ideas why registering a `core safe-update` command requires also registering the command on the `find_command_to_run_pre` hook? It doesn't seem to be safely deferred and instead produces this error:

![image](https://user-images.githubusercontent.com/36432/34133994-ec5332cc-e40c-11e7-92d0-40599e870e6a.png)

Reply:
I cannot reproduce this behavior. Here's what I get when I remove the "hack" to fix it:

<img width="824" alt="image 2017-12-23 at 7 50 37 am" src="https://user-images.githubusercontent.com/83631/34317765-fc06c7ea-e7b5-11e7-940c-35e884ccfd47.png">

Are you running on latest `master` ?

Reply:
Weird. Yes, this was running latest master.

I wonder if there's some race condition involved. Although, it reproduced consistently for me at the time.

I'll close for now and reopen if I run into it again.